typings/index.d.ts

```diff
@@ -1240,7 +1252,7 @@ declare module 'discord.js' {
   export class PermissionOverwrites {
     constructor(guildChannel: GuildChannel, data?: object);
```

Comment:
```suggestion
    constructor(guildChannel: GuildChannelTypes, data?: object);
```
this should also use `GuildChannelTypes`